sw/device/tests/rstmgr_sw_rst_ctrl_test.c

```diff
@@ -0,0 +1,353 @@
+  };
+  CHECK_DIF_OK(dif_i2c_configure(&i2c1, cfg));
+  LOG_INFO("%s update value=0x%x", node->name, read_test_reg(node));
+}
+void i2c2_config(node_t *node) {
+  const dif_i2c_config_t cfg = {
+      .start_signal_setup_cycles = 5525,
+      .start_signal_hold_cycles = 6636,
+  };
+  CHECK_DIF_OK(dif_i2c_configure(&i2c2, cfg));
+  LOG_INFO("%s update value=0x%x", node->name, read_test_reg(node));
+}
+
+// peripheral template
+static node_t test_node[NUM_NODES] = {
+    [RSTMGR] =
+        {
+            .name = "RSTMGR",
+            .base = TOP_EARLGREY_RSTMGR_AON_BASE_ADDR,
+            .init = rstmgr_init,
+        },
+    [SPI_DEVICE] =
+        {
+            .name = "SPI_DEVICE",
+            .base = TOP_EARLGREY_SPI_DEVICE_BASE_ADDR,
+            .offset = SPI_DEVICE_CFG_REG_OFFSET,
+            .init = spi_dev_init,
+            .config = spi_dev_config,
+            .prgm_val = 0x3f0c,
+        },
+    [SPI_HOST0] =
+        {
+            .name = "SPI_HOST0",
+            .base = TOP_EARLGREY_SPI_HOST0_BASE_ADDR,
+            .offset = SPI_HOST_CONFIGOPTS_REG_OFFSET,
+            .init = spi_host0_init,
+            .config = spi_host0_config,
+            .prgm_val = 0x3210000,
+        },
+    [SPI_HOST1] =
+        {
+            .name = "SPI_HOST1",
+            .base = TOP_EARLGREY_SPI_HOST1_BASE_ADDR,
+            .offset = SPI_HOST_CONFIGOPTS_REG_OFFSET,
+            .init = spi_host1_init,
+            .config = spi_host1_config,
+            .prgm_val = 0x6540000,
+        },
+    [USB] =
+        {
+            .name = "USB",
+            .base = TOP_EARLGREY_USBDEV_BASE_ADDR,
+            .offset = USBDEV_EP_OUT_ENABLE_REG_OFFSET,
+            .init = usbdev_init,
+            .config = usbdev_config,
+            .prgm_val = 0xc3,
+        },
+    [USBIF] =
+        {
+            .name = "USBIF",
+            .base = TOP_EARLGREY_USBDEV_BASE_ADDR,
+            .offset = USBDEV_RXENABLE_OUT_REG_OFFSET,
+            .init = usbif_init,
+            .config = usbif_config,
+            .prgm_val = 0x695,
+        },
+    [I2C0] =
+        {
+            .name = "I2C0",
+            .base = TOP_EARLGREY_I2C0_BASE_ADDR,
+            .offset = I2C_TIMING0_REG_OFFSET,
+            .init = i2c0_init,
+            .config = i2c0_config,
+            .prgm_val = 0x8b00cfe,
+        },
+    [I2C1] =
+        {
+            .name = "I2C1",
+            .base = TOP_EARLGREY_I2C1_BASE_ADDR,
+            .offset = I2C_TIMING1_REG_OFFSET,
+            .init = i2c1_init,
+            .config = i2c1_config,
+            .prgm_val = 0x114010d8,
+        },
+    [I2C2] = {.name = "I2C2",
+              .base = TOP_EARLGREY_I2C2_BASE_ADDR,
+              .offset = I2C_TIMING2_REG_OFFSET,
+              .init = i2c2_init,
+              .config = i2c2_config,
+              .prgm_val = 0x19ec1595},
+};
+
+bool test_main(void) {
+  // reset all devices
+  for (int n = RSTMGR; n < NUM_NODES; ++n) {
+    test_node[n].init(&test_node[n]);
```

Comment:
It might be cleaner to register the init function pointer to point to the init DIF directly, then just call it here, wrapped in the CHECK_DIF_OK macro. The you can get rid of all of the `*_init()` functions above (it feels unnecessary to wrap the same init DIF with several functions for each instance of the peripheral).

Reply:
You could move the reading of the reset value to the loop too. Also I would make the LOGs (e.g., L106, and others) a CHECK()

Reply:
rewrote a single init function with switch / macro